paddle/ir/core/parser/ir_parser.cc

```diff
@@ -0,0 +1,317 @@
+    return dialect->ParseType(*this);
+  }
+}
+Attribute IrParser::ParseAttribute() {
+  auto parenthesis_token = ConsumeToken();
+  if (parenthesis_token.val_ == "true" || parenthesis_token.val_ == "false") {
+    return builder->bool_attr(parenthesis_token.val_ == "true");
+  }
+  string attribute_type = PeekToken().val_;
+  if (attribute_type == "String") {
+    ConsumeAToken("String");
+    ConsumeAToken(")");
+    string val = ConsumeToken().val_;
+    return builder->str_attr(val);
+  } else if (attribute_type == "Float") {
+    ConsumeAToken("Float");
+    ConsumeAToken(")");
+    string val = ConsumeToken().val_;
+    return builder->float_attr(atof(val.c_str()));
+  } else if (attribute_type == "Double") {
+    ConsumeAToken("Double");
+    ConsumeAToken(")");
+    string val = ConsumeToken().val_;
+    return builder->double_attr(atof(val.c_str()));
+  } else if (attribute_type == "Int32") {
+    ConsumeAToken("Int32");
+    ConsumeAToken(")");
+    string val = ConsumeToken().val_;
+    return builder->int32_attr(atoi(val.c_str()));
+  } else if (attribute_type == "Int64") {
+    ConsumeAToken("Int64");
+    ConsumeAToken(")");
+    string val = ConsumeToken().val_;
+    return builder->int64_attr(atoll(val.c_str()));
+  } else if (attribute_type == "Array") {
+    ConsumeAToken("Array");
+    ConsumeAToken(")");
+    ConsumeAToken("[");
+    std::vector<Attribute> array_attribute;
+    while (PeekToken().val_ != "]") {
+      array_attribute.push_back(ParseAttribute());
+      if (PeekToken().val_ == "]") break;
+      ConsumeAToken(",");
+    }
+    ConsumeAToken("]");
+    return builder->array_attr(array_attribute);
+  } else {
+    IR_ENFORCE(attribute_type.find('.') != std::string::npos,
+               "No function parsing " + attribute_type + " exists!" +
+                   GetErrorLocationInfo());
+    auto dialect_name = attribute_type.substr(0, attribute_type.find('.'));
+    auto dialect = ctx->GetRegisteredDialect(dialect_name);
+    return dialect->ParseAttribute(*this);
+  }
+}
+
+std::unique_ptr<Program> IrParser::ParseProgram() {
+  std::unique_ptr<Program> program(new Program{ctx});
+  ParseBlock(*program->block());
+  return program;
+}
+
+void IrParser::ParseRegion(Region& region) {  // NOLINT
+  while (PeekToken().token_type_ != EOF_) {
+    Block* block = new Block{};
+    ParseBlock(*block);
+    region.push_back(block);
+  }
+}
+
+void IrParser::ParseBlock(Block& block) {  // NOLINT
+  ConsumeAToken("{");
+  while (PeekToken().val_ != "}") {
+    auto op = ParseOperation();
+    block.push_back(op);
+  }
+  ConsumeAToken("}");
+}
+
+Operation* IrParser::ParseOperation() {
+  std::vector<string> opresultindex = ParseOpResultIndex();
+  ConsumeAToken("=");
+
+  OpInfo opinfo = ParseOpInfo();
+
+  std::vector<OpResult> inputs = ParseOpRandList();
+
+  ir::AttributeMap attributeMap = ParseAttributeMap();
+
+  ConsumeAToken(":");
+  ConsumeAToken("(");
+  while (ConsumeToken().token_type_ != ARRAOW) {
```

Comment:
这里可以这样改吗？感觉这样应该更清晰吧，这样consume的话意图不太容易理解
```suggestion
  ParseFunctionTypeList();
```

Reply:
已修改·